src/core/main.c

```diff
@@ -293,6 +293,36 @@ static int parse_crash_chvt(const char *value) {
         return 0;
 }
 
+static int parse_confirm_spawn(const char *value, char **console) {
+        char *cons;
+        int r;
+
+        r = parse_boolean(value);
+        if (r == 0) {
+                *console = NULL;
+                return 0;
+        }
+        if (r > 0)
+                value = "/dev/console";
+
+        if (strcmp(value, "/dev/console")) {
+                _cleanup_close_ int fd = -1;
+
+                fd = open_terminal(value, O_RDWR|O_NOCTTY|O_CLOEXEC);
+                if (fd < 0) {
+                        log_warning_errno(fd, "failed to open %s: %m, using default console", value);
+                        value = "/dev/console";
+                }
+        }
+
+        cons = strdup(value);
```

Comment:
`free_and_strdup`. Also, if you allocate this sometimes, "/dev/console" should be `free_and_strdup()`ed too.